packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,239 @@
+} from "../revertibles";
+import { SharedString } from "../sharedString";
+import { assertEquivalentSharedStrings } from "./intervalUtils";
+import {
+	Operation,
+	FuzzTestState,
+	RevertOperation,
+	makeReducer,
+	RevertibleSharedString,
+	isRevertibleSharedString,
+} from "./intervalCollection.fuzzUtils";
+import { makeOperationGenerator } from "./intervalCollection.fuzz.spec";
+import { minimizeTestFromFailureFile } from "./intervalCollection.fuzzMinimization";
+
+// Note: none of these options are currently exercised, since the fuzz test fails with pretty much
+// any configuration due to known bugs. Once shared interval collections are in a better state these
+// should be revisited.
+interface OperationGenerationConfig {
+	/**
+	 * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+	 * Note due to concurency, during test execution the actual length of the string may exceed this.
+	 */
+	maxStringLength?: number;
+	/**
+	 * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+	 * Note due to concurency, during test execution the actual number of intervals may exceed this.
+	 */
+	maxIntervals?: number;
+	maxInsertLength?: number;
+	intervalCollectionNamePool?: string[];
+	propertyNamePool?: string[];
+	validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+	maxStringLength: 1000,
+	maxIntervals: 100,
+	maxInsertLength: 10,
+	intervalCollectionNamePool: ["comments"],
+	propertyNamePool: ["prop1", "prop2", "prop3"],
+	validateInterval: 100,
+};
+
+// Since the clients are created by the fuzz harness, the factory object must be
+// modified in order to set the mergeTreeUseNewLengthCalculations option on the
+// underlying merge tree.
+class RevertibleFactory extends SharedStringFactory {
+	options = { mergeTreeUseNewLengthCalculations: true };
+	public async load(
+		runtime: IFluidDataStoreRuntime,
+		id: string,
+		services: IChannelServices,
+		attributes: IChannelAttributes,
+	): Promise<SharedString> {
+		runtime.options.mergeTreeUseNewLengthCalculations = true;
+		return super.load(runtime, id, services, attributes);
+	}
+
+	public create(document: IFluidDataStoreRuntime, id: string): SharedString {
+		document.options.mergeTreeUseNewLengthCalculations = true;
+		return super.create(document, id);
+	}
+}
+
+const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+
+emitter.on("clientCreate", (client) => {
+	const channel = client.channel as RevertibleSharedString;
+	channel.revertibles = [];
+
+	channel.on("createIntervalCollection", (label) => {
+		const collection = channel.getIntervalCollection(label);
+
+		assert(isRevertibleSharedString(channel));
+		collection.on("addInterval", (interval, local, op) => {
+			if (local) {
+				appendAddIntervalToRevertibles(interval, channel.revertibles);
+			}
+		});
+		// Note: delete and change interval edits are disabled for now, and will be reenabled
```

Comment:
in general, prefer switching off large blocks of code with conditionals rather than commenting them out--this makes sure that the code still compiles, and e.g. renames that happen will apply correctly.

Here, I think you should actually be able to leave this code uncommented and without conditionals, and instead disable `deleteInterval` and `changeInterval` events by setting their weights to 0 where you construct the generator.

Reply:
If I set the weights to 0, I still get some of the errors I was hitting before with appendChangeIntervalToRevertibles and revertLocalDelete. I'm not sure why, because if the operations aren't generated, the listeners shouldn't be invoked, but the errors do still happen.

Reply:
That's the reason I ended up setting the addInterval weight to 0